gelf_writer.go

```diff
@@ -189,11 +192,20 @@ func (w *Writer) WriteMessage(m *Message) (err error) {
 
 	var zBuf bytes.Buffer
 	var zw io.WriteCloser
+
 	switch w.CompressionType {
 	case CompressGzip:
-		zw, err = gzip.NewWriterLevel(&zBuf, w.CompressionLevel)
+		if w.gzipWriter == nil {
```

Comment:
Since we have a constructor (`NewWriter`), I suggest initialize in there instead.

Reply:
Actually, no, I just realized `Writer` is exported, so if someone has the great idea of instantiating it directly, `zw` becomes unset and unaccessible at the same time.